src/Actions/Media/UpdateMedia.php

```diff
@@ -22,13 +23,32 @@ public static function models(): array
 
     public function performAction(): Model
     {
+        /** @var Media $media */
         $media = resolve_static(Media::class, 'query')
             ->whereKey($this->data['id'])
             ->first();
 
+        $currentFileName = Str::beforeLast($media->file_name, '.');
+        $paths = [];
+        if (data_get($this->data, 'file_name')) {
+            $this->data['file_name'] = Str::finish($this->data['file_name'], '.' . $media->extension);
+
+            $paths[] = $media->getPath();
+
+            foreach ($media->getGeneratedConversions() as $conversion => $generated) {
+                $paths[] = $media->getPath($conversion);
+            }
+        }
+
         $media->fill($this->data);
         $media->save();
 
+        if ($paths) {
```

Comment:
no need for extra if. If $paths is an empty array the foreach statement does nothing anyways.